plugin/mutate/doc/design/options.md

```diff
@@ -0,0 +1,220 @@
+
+
+- --db : Specify which sqlite3 database to use for the mutation testing (default: dextool_mutate.sqlite3). This option could be used when several databases exists for i.e. different versions of the code, or simply because the user have chosen to create several databases for result.
+
+
+- --out : Path used as the root for mutation/reporting of files (default: .).
+
+
+- --help [-h] : Display the help information in terminal window (less detailed). By writing, for example, *dextool mutate analyze --help*, the user will display the help for the analyze command.
+
+## Admin
+*Admin-mode for the plugin. Is used by to execute administrative commands and to initialize/setup mutation testing for a project.*
+
+
+- --dump-config : Dump the detailed configuration in the terminal. Could be used to create and setup your own configuration-file.
+
+
+- --init : Create an initial configuration to use in the current workspace. This command is used when the mutation testing is setup for a specific project the first time.
+
+
+- --mutant : Mutants to operate on.
+    - *any* : Any mutant generated.
+    - *ror* : Relational Operator Replacement.
+    - *rorp* : Relational Operator Replacement (Pointer).
+    - *lcr* : Logical Connector Replacement.
+    - *lcrb* : Logical Connector Replacement (Bit-wise).
+    - *aor* : Arithmetical Operator Replacement.
+    - *uoi* : Unary Operator Insertion.
+    - *abs* : Absolute Value Insertion.
+    - *sdl* : Statement Deletion.
+    - *cor* : Conditional Operator Replacement.
+    - *dcc* : Decision/Condition Coverage.
+    - *dcr* : Decision/Condition Replacement.
+
+
+- --operation : Administrative operation to perform
+    -   *none* : Performs no operation.
+    -   *resetMutant* : lets the user reset all mutants with the status/state specified with *--status* to the status specified with *--to-status*.
+    -   *removeMutant* : TODO
+    -   *removeTestCase* : TODO
+
+
+- --test-case-regex : Regular expression to use when removing test cases.
+
+
+- --status : Change mutants with this status/state to the value specified by *--to-status-flag*. The typical usage of this option is to reset the mutants with *alive* status to *unknown* in order to conduct mutation testing again after the test suite has been extended.
+    - *unknown* : Mutants that is either untested or caused unknown errors when compiling.
+    - *killed* : Mutants that were detected by the test suite (one or more tests failed).
+    - *alive* : Mutants that were not detected by the test suite (all tests passed).
+    - *killedByCompiler* : Invalid mutants generated that caused the compilation of the project to fail.
+    - *timeout* : Mutants that timed out during test suite execution.
+
+
+- --to-status : Reset mutants to status/state (default: unknown). (see *--status*).
+    - *unknown* : Mutants that is either untested or caused unknown errors when compiling.
```

Comment:
unknown error when compiling can be removed. It is covered by killedByCompiler. A mutant in the state unknown hasn't been "processed" in any way by the tool

Reply:
Hmm, this is code from the function _runTester_ in _plugin/mutate/source/dextool/plugin/mutate/backend/test_mutant/package.d_. Should I maybe formulate it differently?

```
try {
    auto p = spawnSession(compile_p.program ~ compile_p.arguments);
    auto res = p.wait;
    if (res.terminated && res.status != 0)
        return Mutation.Status.killedByCompiler;
    else if (!res.terminated) {
        logger.warning("unknown error when executing the compiler").collectException;
        return Mutation.Status.unknown;
    }
} catch (Exception e) {
    logger.warning(e.msg).collectException;
}
```

Maybe something like:
`Mutants that is either untested or caused unknown errors when trying to execute compilation script.`